docs/basics/101-145-hooks.rst

```diff
@@ -0,0 +1,162 @@
+command, *but only* if the previous :command:`install` command was successful.
+
+Such automation allows for flexible and yet automatic responses to the results
+of DataLad commands, and can be done with DataLad's *result hooks*.
+Generally speaking, `hooks <https://en.wikipedia.org/wiki/Hooking>`__ intercept
+function calls or events and allow to extend the functionality of a program.
+DataLad's result hooks are calls to other DataLad commands after the command
+resulted in a specified result -- such as a successful install.
+
+To understand how hooks can be used and defined, we have to briefly mention
+DataLad's *command result evaluations*. Whenever a DataLad
+command is executed, an internal evaluation generates a *report* on the status
+and result of the command. Internally, this is useful for final result
+rendering, error detection, and logging. However, by using hooks, you can
+utilize these evaluations for your own purposes and "hook" in more commands
+whenever an evaluation fulfills your criteria.
+
+To be able to specify matching criteria, you need to be aware of the potential
+criteria you can match against. The evaluation report is a dictionary with
+``key:value`` pairs. The following table provides an overview on some of the
+available keys and their possible values:
+
+.. list-table::
+   :widths: 50 100
+   :header-rows: 1
+
+   * - Key name
+     - Values
+   * - ``action``
+     - ``get``, ``install``, ``drop``, ``status``, ... (any command's name)
+   * - ``type``
+     - ``file``, ``dataset``, ``symlink``, ``directory``
+   * - ``status``
+     - ``ok``, ``notneeded``, ``impossible``, ``error``
+   * - ``path``
+     - The path the previous command operated on
+
+These key-value pairs provide the basis to define matching rules that -- once met --
+can trigger the execution of custom hooks.
+To define a hook based on certain command results, two configuration variables
+need to be set in ``.datalad/config``:
+
+.. code-block:: bash
+
+   datalad.result-hook.<name>.match-json
+
+and
+
+.. code-block:: bash
+
+   datalad.result-hook.<name>.call-json
+
+Here is what you need to know about these variables:
+
+- The ``<name>`` part of the configurations is the same for both variables, and can be
+  an arbitrarily [#f1]_ chosen name that serves as an identifier for the hook you are
+  defining.
+
+- The first configuration variable, ``datalad.result-hook.<name>.match-json``, defines
+  the requirements that a result evaluation needs to match in order to trigger the hook.
+
+- The second configuration variable, ``datalad.result-hook.<name>.call-json``, defines
+  what the hook execution comprises. It can be any DataLad command of your choice.
+
+And here is how to set the values for these variables:
+
+- The value for ``datalad.result-hook.<name>.match-json`` needs to be specified as
+  a JSON-encoded dictionary with any number of keys, such as
+
+  .. code-block:: bash
+
+     {"type": "file", "action": "get", "status": "notneeded"}
```

Comment:
We should be more specific that this is the plain JSON, but written in the config file it must look different:

```
mih@meiner ~ % git config --file my.conf --add dummy.match-json '{"type": "file", "action": "get", "status": "notneeded"}'
mih@meiner ~ % cat my.conf
[dummy]
        match-json = {\"type\": \"file\", \"action\": \"get\", \"status\": \"notneeded\"}
```

Reply:
Ah, I see now this is done below. So maybe just somehow point out that this is not the final form in the config file.